File: Arrays & Strings/2dcolumn.cpp
#include <iostream>
using namespace std;
int main()
{
    int rows, columns;
    cout << "How many rows and columns ?";
    cin >> rows >> columns;
    cout << "Enter your rows and columns data ->" << endl;
    int twodim[rows][columns];
    // Input
    for (int i = 0; i < rows; i++)
    {
        for (int j = 0; j < columns; j++)
        {
            cout << "Enter row " << i << " column " << j << " :";
            cin >> twodim[i][j];
            cout << endl;
        }
    }
    // column Wise PRINT
    for (int i = 0; i < columns; i++)
    {
        for (int j = 0; j < rows; j++)
        {
            cout << "|" << twodim[j][i] << "|";
        } //                       |--|->Swap positions of (i) with (j)
        cout << endl;
    }
    return 0;
}

File: Arrays & Strings/2dcolumn.java
import java.util.Scanner;

class column {
    public static void main(String[] args) {
        int rows, columns;
        System.out.println("How many rows and columns");
        Scanner scan = new Scanner(System.in);
        rows = scan.nextInt();
        columns = scan.nextInt();
        int[][] twodim = new int[rows][columns];
        // Input
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns; j++) {
                System.out.print("Enter Row" + i + " Columns " + j);
                twodim[i][j] = scan.nextInt();
                System.out.print("\n");
            }
        }
        // Column wise print
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < columns; j++) {
                System.out.print("|" + twodim[j][i] + "|");
            }
            System.out.print("\n");
        }
        scan.close();
    }
}